bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableConfiguration.java

```diff
@@ -47,42 +49,87 @@
   }
 
+  private static Configuration setConnectionClass(Configuration config) {
+    Preconditions.checkState(CONNECTION_CLASS != null,
+        "Could not find an appropriate BigtableConnection class");
+    config.set(HConnection.HBASE_CLIENT_CONNECTION_IMPL, CONNECTION_CLASS.getCanonicalName());
+    return config;
+  }
+
   /**
-   * <p>configure.</p>
+   * <p>Create a configuration with HBase resources loaded and Connection Class set.</p>
+   */
+  public static Configuration create() {
+    return setConnectionClass(HBaseConfiguration.create());
+  }
+
+  /**
+   * <p>Create a configuration with HBase resources loaded and Defaults set.</p>
    *
    * @param projectId a {@link java.lang.String} object.
    * @param instanceId a {@link java.lang.String} object.
    * @return a {@link org.apache.hadoop.conf.Configuration} object.
    */
-  public static Configuration configure(String projectId, String instanceId) {
-    Configuration config = new Configuration(false);
+  public static Configuration create(String projectId, String instanceId) {
+    return configure(create(), projectId, instanceId);
+  }
+
+  /**
+   * <p>Configure and return a configuration with HBase resources loaded and Defaults set.</p>
+   *
+   * @param config a {@link org.apache.hadoop.conf.Configuration} object.
+   * @param projectId a {@link java.lang.String} object.
+   * @param instanceId a {@link java.lang.String} object.
+   * @return the modified {@link org.apache.hadoop.conf.Configuration} object.
+   */
+  public static Configuration configure(Configuration config, String projectId, String instanceId) {
+    setConnectionClass(config);
     config.set(BigtableOptionsFactory.PROJECT_ID_KEY, projectId);
     config.set(BigtableOptionsFactory.INSTANCE_ID_KEY, instanceId);
     return config;
   }
 
   /**
+   * <p>Create a configuration with HBase resources loaded and Defaults set.</p>
+   *
+   * Use {link #create(String, String)} instead.
+   *
+   * @param projectId a {@link java.lang.String} object.
+   * @param instanceId a {@link java.lang.String} object.
+   * @return a {@link org.apache.hadoop.conf.Configuration} object.
+   */
+  @Deprecated
+  public static Configuration configure(String projectId, String instanceId) {
+    return create(projectId, instanceId);
+  }
+
+  /**
    * <p>connect.</p>
    *
    * @param projectId a {@link java.lang.String} object.
    * @param instanceId a {@link java.lang.String} object.
    * @return a {@link org.apache.hadoop.hbase.client.Connection} object.
    */
+  @Deprecated
   public static Connection connect(String projectId, String instanceId) {
-    return connect(configure(projectId, instanceId));
+    return connect(create(projectId, instanceId));
   }
 
   /**
    * <p>connect.</p>
    *
-   * @param conf a {@link org.apache.hadoop.conf.Configuration} object.
+   * Use {@link org.apache.hadoop.hbase.client.ConnectionFactory#createConnection(Configuration)} if
+   * you want predictable code.
```

Comment:
I think that this comment needs to be updated.  #createConnection is a fine alternative, but it needs HConnection.HBASE_CLIENT_CONNECTION_IMPL.  The intent here is to bypass that, and have a simple way to create a BigtableConnection object.

It's unclear to me what "if you want predictable code" means in this context.